tests/ert3/stats/test_distributions.py

```diff
@@ -59,3 +61,66 @@ def test_gaussian_distribution_invalid():
     err_msg_both = "Cannot create gaussian distribution with both size and index"
     with pytest.raises(ValueError, match=err_msg_both):
         ert3.stats.Gaussian(0, 1, size=10, index=list(range(10)))
+
+
+@pytest.mark.parametrize(
+    ("size", "lower_bound", "upper_bound"),
+    (
+        (10000, 0, 1),
+        (20000, 10, 20),
+    ),
+)
+def test_uniform_distribution(size, lower_bound, upper_bound):
+    uniform = ert3.stats.Uniform(lower_bound, upper_bound, size=size)
+
+    prev_samples = set()
+    for _ in range(10):
+        sample = uniform.sample()
+
+        assert len(sample) == size
+
+        assert tuple(sample) not in prev_samples
+        prev_samples.add(tuple(sample))
+
+        assert sample.min() == pytest.approx(lower_bound, abs=_TOLERANCE)
+        assert sample.mean() == pytest.approx(
+            (lower_bound + upper_bound) / 2, abs=_TOLERANCE
+        )
+        assert sample.max() == pytest.approx(upper_bound, abs=_TOLERANCE)
+
+
+@pytest.mark.parametrize(
```

Comment:
Perhaps this could be better done with hypothesis

```
from hypothesis import given
import hypothesis.strategies as st

uniform_distribution_indecies = 
uniform_distributions = st.build(
  ert3.stats.Uniform,
  st.integers(min_value=0,max_value=2),
  st_integers(min_value=3, max_value=5),
  index=st.just(tuple("a" * i for i in range(1, 10)))

@given(uniform_distributions)
def test_uniform_distribution_is_sorted(uniform):
    assert sorted(uniform.sample().keys() == sorted(uniform._index)
```

Reply:
Thanks for suggesting hypothesis, I will have a look 👍

Reply:
Let's leave this for #1231